builder/docker_test.go

```diff
@@ -0,0 +1,170 @@
+		},
+		{
+			Filepath:        "some/path/to/doesnotexist",
+			ExpectedDirPath: "some/path/to",
+		},
+
+		// Error cases.
+		{Filepath: "not/a/real/path", ShouldFail: true},
+	}
+
+	testDir, err := ioutil.TempDir("", "getdirfrompath-test-")
+	if err != nil {
+		t.Fatalf("couldn't create temporary directory to write test cases: %s", err)
+	}
+	defer func() {
+		_ = os.RemoveAll(testDir)
+	}()
+
+	// Set up the dirs/files for testing
+	path := filepath.Join(testDir, "some/path/to/directory")
+	if err = os.MkdirAll(path, 0777); err != nil {
+		t.Errorf("Failed to create test directory: %q\n%s", path, err)
+	}
+	path = filepath.Join(testDir, "some/path/to/file") // "some/path/to" already exists from previous command
+	f, err := os.Create(path)
+	if err != nil {
+		t.Errorf("Failed to create test file: %q\n%s", path, err)
+	}
+	defer func() {
+		_ = f.Close()
+	}()
+
+	for _, tt := range tests {
+		fullPath := filepath.Join(testDir, tt.Filepath)
+
+		dir, err := getDirFromConfigPath(fullPath)
+		dir = strings.TrimPrefix(dir, testDir+"/")
+		failed := err != nil
+		if failed != tt.ShouldFail {
+			if tt.ShouldFail {
+				t.Errorf("unexpected success when parsing filepath\nFILEPATH: %s\nPARSED DIR: %s\n", tt.Filepath, dir)
+			} else {
+				t.Errorf("unexpected error parsing bundle: %s\nFILEPATH: %s\nEXPECTED DIR: %s\n", err, tt.Filepath, tt.ExpectedDirPath)
+			}
+			continue
+		}
+		if tt.ShouldFail {
+			continue
+		}
+
+		if !reflect.DeepEqual(dir, tt.ExpectedDirPath) {
+			t.Errorf("got wrong dir when parsing filepath\nFILEPATH: %s\nPARSED DIR: %s\nEXPECTED DIR: %s", tt.Filepath, dir, tt.ExpectedDirPath)
+		}
+	}
+}
+
+func TestCanAccess(t *testing.T) {
+	tests := []struct {
+		Filepath   string
+		ShouldFail bool
+	}{
+		{Filepath: "shouldwork"},
+
+		// Error cases.
+		{Filepath: "noread", ShouldFail: true},
+		{Filepath: "nowrite", ShouldFail: true},
+		{Filepath: "noreadwrite", ShouldFail: true},
+		{Filepath: "/usr/var", ShouldFail: true},
+	}
+
+	testDir, err := ioutil.TempDir("", "canaccess-test-")
+	if err != nil {
+		t.Fatalf("couldn't create temporary directory to write test cases: %s", err)
+	}
+	defer func() {
+		_ = os.RemoveAll(testDir)
+	}()
+
+	// Set up the dirs/files for testing
+	path := filepath.Join(testDir, "shouldwork")
+	if err := os.Mkdir(path, 0777); err != nil {
+		t.Errorf("Failed to create test directory: %q\n%s", path, err)
+	}
+	path = filepath.Join(testDir, "noread")
+	if err := os.Mkdir(path, 0333); err != nil {
+		t.Errorf("Failed to create test directory: %q\n%s", path, err)
+	}
+	path = filepath.Join(testDir, "nowrite")
+	if err := os.Mkdir(path, 0555); err != nil {
+		t.Errorf("Failed to create test directory: %q\n%s", path, err)
+	}
+	path = filepath.Join(testDir, "noreadwrite")
+	if err := os.Mkdir(path, 0111); err != nil {
+		t.Errorf("Failed to create test directory: %q\n%s", path, err)
+	}
+
+	for _, tt := range tests {
+		fullPath := tt.Filepath
+		if !strings.HasPrefix(fullPath, "/") {
```

Comment:
As mentioned above I'd just drop the `/usr/var` check and then drop the `fullPath` variable and checks.